sklearn/model_selection/_split.py

```diff
@@ -1472,16 +1467,8 @@ class GroupShuffleSplit(ShuffleSplit):
 
     '''
 
-    def __init__(self, n_splits=5, test_size="default", train_size=None,
+    def __init__(self, n_splits=5, test_size=0.2, train_size=None,
                  random_state=None):
-        if test_size == "default":
-            if train_size is not None:
-                warnings.warn("From version 0.21, test_size will always "
-                              "complement train_size unless both "
-                              "are specified.",
-                              FutureWarning)
-            test_size = 0.2
```

Comment:
This change (setting `test_size=0.2` instead of `test_size=None  in the constructor) does not match the `FutureWarning` contract:

```python
GroupShuffleSplit(train_size=0.5)
```

should yield `test_size=0.5 == 1. - train_size` and not `test_size=0.2`.

Fixing this will mean that we need to pass a different value for `default_test_size` for `ShuffleSplit` / `StratifiedShuffleSplit` (`default_test_size=0.1`) than the value used by default for `GroupShuffleSplit` (`default_test_size=0.2`)...

This means that we also need a new test for this case.

Reply:
There's no way to match the `FutureWarning` contract.
How do you solve that:
```python
ShuffleSplit(train_size=.5) => (train=50%, test=50%)
ShuffleSplit(train_size=.5, test_size=.2) => (train=50%, test=20%)
```
The only way is that the default value is None and None is interpreted as 0.2 only if `train_size` is None.

Reply:
> The only way is that the default value is None and None is interpreted as 0.2 only if train_size is None.

Yes. What is the problem?

Reply:
Oh I must have misunderstood because I thought that was the problem you mentionned. What is the problem then ? it was just the missing tests ?

Reply:
Maybe I am not clear but I think the current code does not implement what `FutureWarning` promised we would do.

Here is a more detailed case of the expected behavior:

https://github.com/scikit-learn/scikit-learn/pull/13483/commits/300e0fc84e5d41ce32a798de1c097a0884e35902#r268222280

Basically we would like the `test_size` parameter to behave in `GroupShuffleSplit` as it behaves in `ShuffleSplit` but with a different default value for `test_size` (unfortunately).

To recap:

- `train_test_split(train_size=None, test_size=None)` means `test_size=0.25`
- `(Stratified)ShuffleSplit(train_size=None, test_size=None)` means `test_size=0.1`
- `GroupShuffleSplit(train_size=None, test_size=None)` means `test_size=0.2`

In all cases, when `train_size is not None` and `test_size=None`, we should always have:

     test_size == 100% - train_size

Reply:
You're right. I thought I took care of all of them but I messed up the GroupShuffleSplit.
In the end I set the default test size as a private attribute of the classes.